daffodil-tdml-lib/src/main/scala/org/apache/daffodil/tdml/TDMLRunner.scala

```diff
@@ -2364,6 +2361,35 @@ sealed abstract class DocumentPart(part: Node, parent: Document) {
     }
   }.trim.toUpperCase()
 
+  /*
+  * Description: Compare ByteDocumentPart/BitsDocumentPart string to a string of valid characters.
+  *              If the string runs into non-valid characters, skip over them.
+  *              If the string runs into "//", any character following it will be skipped (including valid ones).
+  *              Characters on a new line will be considered again, unless they are found after a "//"
+  * Parameters: String containing the acceptable characters (0,1 for bits) (0-9,A-F,a-f for bytes)
```

Comment:
The parameter arg contains not just acceptable chars, but also can be comment chars or comment syntax introduced by //.

Reply:
I'd also suggest we pass in the string to remove comments from (e.g. rawPartContent), rather than requiring that it be a member of the class that the function accesses. That way this is more of a utility function (and could even go in some Utils object) and not something specific to certain classes that need a certain member variable.

Reply:
Yes, I also prefer functional implementations where you pass in everything, they return everything to you as a result, and they don't modify any mutable state (unless you pass it in as an argument too).